thewalrus/quantum/conversions.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2019-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Functions for transforming one type of covariance-matrix-like object into
+another as well as reducing a Gaussian state.
+"""
+
+import numpy as np
+
+
+def reduced_gaussian(mu, cov, modes):
+    r""" Returns the vector of means and the covariance matrix of the specified modes.
+
+    Args:
+        mu (array): a length-:math:`2N` ``np.float64`` vector of means.
+        cov (array): a :math:`2N\times 2N` ``np.float64`` covariance matrix
+            representing an :math:`N` mode quantum state.
+        modes (int of Sequence[int]): indices of the requested modes
+
+    Returns:
+        tuple (means, cov): where means is an array containing the vector of means,
+        and cov is a square array containing the covariance matrix.
+    """
+    N = len(mu) // 2
+
+    # reduce rho down to specified subsystems
+    if isinstance(modes, int):
+        modes = [modes]
+
+    if np.any(np.array(modes) > N):
+        raise ValueError("Provided mode is larger than the number of subsystems.")
+
+    if len(modes) == N:
+        # reduced state is full state
+        return mu, cov
+
+    ind = np.concatenate([np.array(modes), np.array(modes) + N])
+    rows = ind.reshape(-1, 1)
+    cols = ind.reshape(1, -1)
+
+    return mu[ind], cov[rows, cols]
+
+
+def Xmat(N):
```

Comment:
Minor: is it worth making the plunge and making these lower-case like python functions conventionally are?

Reply:
I thought about this as well, although this would make them a bit difficult to read. `amat` doesn't really look too good. Maybe something like `a_mat`, `x_mat`, etc. What do you think @nquesada?